RecoEgamma/EgammaTools/plugins/HGCalElectronFilter.cc

```diff
@@ -0,0 +1,180 @@
+// -*- C++ -*-
+//
+// Package:    EgammaTools/HGCalElectronFilter
+// Class:      HGCalElectronFilter
+//
+/**\class HGCalElectronFilter HGCalElectronFilter.cc EgammaTools/HGCalElectronFilter/plugins/HGCalElectronFilter.cc
+
+ Description: filtering of duplicate HGCAL electrons
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Florian beaudette
+//         Created:  Mon, 06 Nov 2017 21:49:54 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectronFwd.h"
+
+//
+// class declaration
+//
+
+class HGCalElectronFilter : public edm::stream::EDProducer<> {
+   public:
+      explicit HGCalElectronFilter(const edm::ParameterSet&);
+      ~HGCalElectronFilter() override;
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+   private:
+      void beginStream(edm::StreamID) override;
+      void produce(edm::Event&, const edm::EventSetup&) override;
+      void endStream() override;
+
+      //virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+      //virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+      //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+      //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+
+      // ----------member data ---------------------------
+      edm::EDGetTokenT<edm::View<reco::GsfElectron>> ElectronsToken_;
```

Comment:
start local variables and data members with lower case letter. Capitalization is reserved for class names.